crypto/keys/types.proto

```diff
@@ -0,0 +1,40 @@
+syntax = "proto3";
+package cosmos_sdk.crypto.keys.v1;
+
+import "third_party/proto/gogoproto/gogo.proto";
+
+option go_package = "github.com/cosmos/cosmos-sdk/crypto/keys";
+
+option (gogoproto.goproto_getters_all)     = false;
+option (gogoproto.goproto_stringer_all)    = false;
+option (gogoproto.goproto_enum_prefix_all) = false;
+
+// Curve defines the key curve types that are available for key generation.
+enum Curve {
+  // Unspecified curve. Returns error on key generation.
+  UNSPECIFIED = 0;
+  // Ed25519 curve
+  ED25519 = 1;
+  // Secp256k1 curve
+  SECP256K1 = 2;
+  // Sr25519 curve
+  SR25519 = 3;
+}
+
+// CompactBitArray is an implementation of a space efficient bit array.
+// This is used to ensure that the encoded data takes up a minimal amount of
+// space after amino encoding.
+// This is not thread safe, and is not intended for concurrent usage.
+message CompactBitArray {
+  bytes extra_bits = 1
+      [(gogoproto.customname) = "ExtraBitsStored", (gogoproto.moretags) = "yaml:\"extra_bits\""];
+  bytes bits = 2 [(gogoproto.customname) = "Elems", (gogoproto.moretags) = "yaml:\"bits\""];
+}
+
+// Multisignature is used to represent the signature object used in the multisigs.
+// Sigs is a list of signatures, sorted by corresponding index.message PrivKeyEd25519 {
+message Multisignature {
+  option (gogoproto.goproto_stringer) = true;
+  CompactBitArray bitarray            = 1 [(gogoproto.customname) = "BitArray"];
```

Comment:
What role does the bitarray play here?

Reply:
you can read more about this in the multisig ADR here: https://github.com/tendermint/tendermint/blob/master/docs/architecture/adr-019-multisigs.md#bit-array